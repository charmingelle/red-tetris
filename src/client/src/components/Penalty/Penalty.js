import React from 'react';
import { Square } from '../Square';
import {
  BOTTOM_LIMIT,
  RIGHT_LIMIT,
  PENALTY_COLOR,
  INVISIBLE_ROW_AMOUNT,
} from '../../constants';

export const Penalty = ({ startRow, squareWidth }) => {
  const penaltyRows = [];

  for (let rowIndex = startRow; rowIndex < BOTTOM_LIMIT; rowIndex++) {
    for (let colIndex = 0; colIndex < RIGHT_LIMIT; colIndex++) {
      penaltyRows.unshift(
        <Square
          left={colIndex}
          top={rowIndex - INVISIBLE_ROW_AMOUNT}
          color={PENALTY_COLOR}
          key={`${rowIndex}+${colIndex}`}
          width={squareWidth}
          borderWidth={squareWidth / 8}
        />,
      );
    }
  }
  return penaltyRows;
};
