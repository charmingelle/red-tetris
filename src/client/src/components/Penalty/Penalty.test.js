import React from 'react';
import { Penalty } from './Penalty';
import renderer from 'react-test-renderer';
import {
  BOTTOM_LIMIT,
  SQUARE_WIDTH,
  OTHER_SQUARE_WIDTH,
} from '../../constants';

test('No penalty', () => {
  expect(
    renderer
      .create(<Penalty startRow={BOTTOM_LIMIT} squareWidth={SQUARE_WIDTH} />)
      .toJSON(),
  ).toMatchSnapshot();
});

test('One penalty row', () => {
  expect(
    renderer
      .create(
        <Penalty
          startRow={BOTTOM_LIMIT - 1}
          squareWidth={OTHER_SQUARE_WIDTH}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
