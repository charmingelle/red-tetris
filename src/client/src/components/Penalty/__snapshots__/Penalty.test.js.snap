// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No penalty 1`] = `null`;

exports[`One penalty row 1`] = `
Array [
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "72px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "64px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "56px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "48px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "40px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "32px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "24px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "16px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "8px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#666666",
        "borderBottom": "1px solid #333",
        "borderLeft": "1px solid #999",
        "borderRight": "1px solid #333",
        "borderTop": "1px solid  #999",
        "height": "8px",
        "left": "0px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
]
`;
